metricbeat/helper/prometheus/textparse.go

```diff
@@ -593,7 +606,10 @@ func ParseMetricFamilies(b []byte, contentType string, ts time.Time) ([]*MetricF
 			if hasExemplar := parser.Exemplar(&e); hasExemplar {
 				exm = &e
 			}
-			lookupMetricName, metric = histogramMetricName(metricName, v, lset.Get(labels.BucketLabel), lbls.String(), &t, false, exm, histogramsByName)
+			lookupMetricName, metric = histogramMetricName(metricName, v, qv, lbls.String(), &t, false, exm, histogramsByName)
+			if metric == nil { // metric name does not have a suffix supported for the type histogram
```

Comment:
may be worth adding a doc to the prometheus histogram definition - https://prometheus.io/docs/concepts/metric_types/#histogram

Reply:
Thanks! I added it above the function used, to be side by side with OpenMetrics: https://github.com/elastic/beats/blob/561dfd74290523134f0f9e5a82d86e15d83399bc/metricbeat/helper/prometheus/textparse.go#L402.